backend/main.go

```diff
@@ -27,16 +41,30 @@ func getFellowHandler(w http.ResponseWriter, req *http.Request) {
 	authorized := isAuthorized(w, req)
 
 	if !authorized {
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusUnauthorized)
+		res := response{
+			Status: "error",
+			Body:   "Incorrect secret given, you are not authorized to use this API",
+		}
+		json.NewEncoder(w).Encode(res)
+
 		endPoint := fmt.Sprintf("/getfellow/%s", vars["username"])
 		logCall("POST", endPoint, "401", startTime)
-		fmt.Fprintf(w, "You are not authorized to use this API")
 		return
 	}
 
 	if vars["username"] == "" {
```

Comment:
I guess a more general `ValidUsername` would be nice. We wouldn't even need to "spend" api calls. Just reach out to `https://github.com/{username}` and checkout the status code (`404` if doesn't exist and `200` otherwise). Then `ValidUsername` would also check for the empty string.

Reply:
I'l include that small check right now